tasks/package.js

```diff
@@ -61,29 +61,101 @@ module.exports = function(grunt) {
 
   grunt.registerTask('package', 'Package the operational codebase for deployment. Use package:compress to create an archive.', function() {
     grunt.loadNpmTasks('grunt-contrib-copy');
+    grunt.loadNpmTasks('grunt-shell');
 
     var config = grunt.config.get('config.packages');
-    var srcFiles = ['**', '!**/.gitkeep'].concat((config && config.srcFiles && config.srcFiles.length) ? config.srcFiles : '**');
+    var srcFiles = (config && config.srcFiles && config.srcFiles.length) ? config.srcFiles : [];
     var projFiles = (config && config.projFiles && config.projFiles.length) ? config.projFiles : [];
 
     // Look for a package target spec, build destination path.
     var packageName = grunt.option('name') || config.name || 'package';
-    var destPath = grunt.config.get('config.buildPaths.packages') + '/' + packageName;
+
+    // Determine if we are using rsync and vmPath and tarball for performance.
+    var useRsync = grunt.config('config.packages.rsync') || false;
+    // Determine if we are creating a tarball archive.
+    var archive = grunt.config('config.packages.archive') || useRsync;
+
+    // When using rsync, generate to a path that can be mounted in the VM.
+    var vmPath = grunt.config.get('config.packages.vmData') || 'build/vm_data';
+    var destPath = vmPath + '/' + packageName;
+    var finalPath = grunt.config.get('config.buildPaths.packages') + '/' + packageName;
+    if (!useRsync) {
+      // If not using rsync, then directly generate to final path.
+      destPath = finalPath;
+    }
+
     var tasks = [];
+    var cleanPaths = [destPath];
+    if (destPath !== finalPath) {
+      cleanPaths.push(finalPath);
+    }
     grunt.option('package-dest', destPath);
+    grunt.config.set('clean.packages', cleanPaths);
+    tasks.push('clean:packages');
+
+    var excludePaths = ['bower_components', 'node_modules', '.gitkeep'];
+
+    if (useRsync) {
```

Comment:
The code I added makes them compatible.  You still use srcFiles, projFiles, and the new excludePaths.  The code builds the proper arguments for either copy or rsync.  So yes, Windows cannot use the "rsync=true" option.

If there is a simple way to detect if rsync is installed I'd be happy to use that to set the default and have it fallback.